src/Raven.Server/Documents/Versioning/VersioningStorage.cs

```diff
@@ -310,15 +310,16 @@ private VersioningConfigurationCollection GetVersioningConfiguration(CollectionN
             if ((nonPersistentFlags & NonPersistentDocumentFlags.FromSmuggler) == NonPersistentDocumentFlags.FromSmuggler)
                 return;
 
-            if (configuration.MaxRevisions.HasValue == false || configuration.MaxRevisions.Value == int.MaxValue)
+            if (configuration.MinimumRevisionsToKeep.HasValue == false &&
+                configuration.MinimumRevisionAgeToKeep.HasValue == false)
                 return;
 
-            var numberOfRevisionsToDelete = revisionsCount - configuration.MaxRevisions.Value;
+            var numberOfRevisionsToDelete = revisionsCount - configuration.MinimumRevisionsToKeep ?? long.MaxValue;
```

Comment:
Wrong here, we can have `MinimumRevisionsToKeep` that is null, but we still need to delete documents that are older than the `MinimumRevisionAgeToKeep`